lib/libzfs/libzfs_dataset.c

```diff
@@ -2921,6 +2921,26 @@ zfs_prop_get(zfs_handle_t *zhp, zfs_prop_t prop, char *propbuf, size_t proplen,
 		zcp_check(zhp, prop, val, NULL);
 		break;
 
+	case ZFS_PROP_SNAPSHOTS_CHANGED:
```

Comment:
This case looks almost identical to `ZFS_PROP_CREATION` except it uses `get_numeric_property()`.  Couldn't `getprop_uint64()` be used here instead to allow the cases to be merged?

Reply:
I used `get_numeric_property()` here to check if the property exists. We should return -1 if the property does not exist or if it is zero, otherwise user will see `Jan 1, 1970` as timestamp.

We can however move it above `ZFS_PROP_CREATION`, and let it fall-through to re-use the code from `ZFS_PROP_CREATION` after the return. But in that scenario, we'd be retrieving the value twice, once to check if it exists and again in `ZFS_PROP_CREATION` via `getprop_uint64()`.

Reply:
Got it.  Then I think keeping as you have it is fine.